### DISCUSSION TITLE
Any guidance or is it even possible to deploy this in private only vpc?

### DISCUSSION THREAD

**Discussion by scobatch** [aws-solutions/workload-discovery-on-aws#516] [Q&A] [answered]
I am attempting to stand this solution up in a dedicated account that already has a private vpc with 2 private subnets.  And the solution is failing on the Gremlin AppSync Stack.   There really isn't a clear error that I am seeing but I suspect this is due to this stack needing to reach out to the internet.
I believe I have setup all the proper endpoints as it instructs in the documentation - although the docs do not give clear guidance on exactly which endpoints are needed.

Any help is greatly appreciated.

**Comment by svozza** [comment, marked as answer]
What's the CloudFormation error in the Gremlin stack you're getting? It's odd, because that stack shouldn't need access to the internet. I have seen problems deploying the OpenSearch stack because there is a custom resource that creates an index and custom resources require access to S3 to complete and the deployment just hangs if it can't

The issue you're most likely to encounter is more after deployment, the discovery process running in that VPC is going to need access to all the APIs documented [here](https://docs.aws.amazon.com/solutions/latest/workload-discovery-on-aws/aws-apis.html) so you'll need to set up PrivateLink endpoints.

**Reply by scobatch** [reply]
Thanks much.  I somehow missed the additional API documentation.   I was able to get the solution to deploy after double checking and adding a few endpoints.

**Reply by svozza** [reply]
Good stuff!

**Comment by aws-ymson** [comment]
Closing this out: Workload Discovery on AWS has been deprecated as of 08/14/2026 and is no longer being maintained.
